### PR TITLE
refactor: convert for loops into list and dict comprehensions

### PR DESCRIPTION
**Pull request by Uzaaft**
# Description

The aim of this PR is to refactor all possible for-loops where lists and dicts are created into list-comprehensions, to get an easy gain of performance.

# How has this been tested?

The list-comprehensions does not add any additional logic, it just required a rewrite of the old logic. If tests passed before, they will(and does locally), pass. 
* Provide instructions so we can reproduce. 
* Please also list any relevant details for your test configuration.

# Checklist:

**None of these are required since this is just a refactor**
- [ ] Update [our Hugo documentation](https://openbb-finance.github.io/OpenBBTerminal/) following [these guidelines](https://github.com/OpenBB-finance/OpenBBTerminal/tree/main/website).
- [ ] Update our tests following [these guidelines](https://github.com/OpenBB-finance/OpenBBTerminal/tree/main/tests).
- [ ] Make sure you are following our [CONTRIBUTING guidelines](https://github.com/OpenBB-finance/OpenBBTerminal/blob/main/CONTRIBUTING.md).
- [ ] If a feature was added make sure to add it to the corresponding [scripts file](https://github.com/OpenBB-finance/OpenBBTerminal/tree/main/scripts).

# Others
- [x] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] My code passes all the checks pylint, flake8, black, ... To speed up development you should run `pre-commit install`.
- [ ] New and existing unit tests pass locally with my changes. You can test this locally using `pytest tests/...`.

### CODE REVIEW

File: openbb_terminal/econometrics/econometrics_view.py [hunk 1]
```diff
@@ -368,10 +368,7 @@ def display_cointegration_test(
     """
 
     if isinstance(datasets, pd.DataFrame):
-        new_datasets = {}
-        for column in datasets.columns:
-            new_datasets[column] = datasets[column]
-
+        new_datasets = {column: datasets[column] for column in datasets.columns}
```
Comment (Uzaaft) [inline openbb_terminal/econometrics/econometrics_view.py:371]: Regarding this:
Why is this here at all? This causes the flake8 lint to fail

Comment (Uzaaft) [reply openbb_terminal/econometrics/econometrics_view.py:371]: The variable `new_datasets` is not being used later on after the if-test

Comment (Uzaaft) [reply openbb_terminal/econometrics/econometrics_view.py:371]: @DidierRLopes  Do you know?
I'm inclined towards moving the unused variable. There seems to be no usage of it in the code.

Comment (DidierRLopes) [reply openbb_terminal/econometrics/econometrics_view.py:371]: This was done by @JerBouma, probably he can shed some insight here

Comment (Uzaaft) [reply openbb_terminal/econometrics/econometrics_view.py:371]: I think a lot of these will come forward as I open PR's for the different refactors btw. 😅

Comment (Uzaaft) [reply openbb_terminal/econometrics/econometrics_view.py:371]: @JerBouma  Could you shed some insight regarding the usage of the unused new_datasets variable?

Comment (JerBouma) [reply openbb_terminal/econometrics/econometrics_view.py:371]: Seems to be legacy code as it is indeed now unused. There have been multiple refractors done to the code in which this got optimized apparently.

Comment (Uzaaft) [reply openbb_terminal/econometrics/econometrics_view.py:371]: Thanks a lot @JerBouma!
@DidierRLopes Perhaps, going forward, if a variable is unused, we remove it? Unless anything else is explicitly mentioned on the code?

Comment (piiq) [reply openbb_terminal/econometrics/econometrics_view.py]: @Uzaaft yes, definitely unused variables should not be in the code. I wonder how that one passed the linters.

Would you be up for jumping into a call on Discord or elsewhere to coordinate the contributing efforts? This would make it easier and quicker to merge PRs

Comment (Uzaaft) [reply openbb_terminal/econometrics/econometrics_view.py]: Yeah sure thing @piiq.  
After the merging that you did, I saw a few more possibilities of using list-comprehensions, so I am including those as well now.

Comment (piiq) [reply openbb_terminal/econometrics/econometrics_view.py]: @Uzaaft can you add me on discord please? `pll_llq#8920`

Comment (Uzaaft) [reply openbb_terminal/econometrics/econometrics_view.py]: I just sent you an request @piiq

**Comment by Uzaaft** [comment]
@piiq All checks passed , but I need to update the branch once again😂

**Review by piiq — APPROVED** [review]
🚀

**Comment by Uzaaft** [comment]
On to the Next one🤓
